contributors/design-proposals/pod-preemption.md

```diff
@@ -0,0 +1,188 @@
+-  Scheduler looks at pod D in the queue. Pod D can fit on Node 2.
+-  Scheduler goes ahead and binds pod D to Node 2 while pods A and B are in their graceful termination period and pod C is not bound yet.
+
+### Example 4
+
+![image](images/preemption_4.png)
+
+-  Everything is similar to example 1, but while scheduler is waiting for pods A and B to gracefully terminate, a new higher priority pod F is created and goes to the head of the queue.
+-  Scheduler evaluates the feasibility of pod F and determines that it can be scheduled on Node 1. So, it sets the future node of pod F to Node 1.
+-  Scheduler looks at pod C and assumes that pod F is already bound to Node 1, so pod C can no longer run on Node 1. Scheduler clears the future node of pod C.
+-  Eventually when pods A and B terminate, pod F is bound to Node 1 and pods C and D remain pending.
+
+# Alternatives Considered
+
+## Rescheduler or Kubelet performs preemption
+
+There are two potential alternatives for the component that performs preemption: rescheduler and Kubelet.  
+Kubernetes has a "[rescheduler](https://kubernetes.io/docs/concepts/cluster-administration/guaranteed-scheduling-critical-addon-pods/)" that performs a rudimentary form of preemption today. The more sophisticated form of preemption that is proposed in this document would require many changes to the current rescheduler. The main drawbacks of using the rescheduler for preemption are
+
+-  requires replicating the scheduler logic in another component. In particular, the rescheduler is responsible for choosing which node the pending pod should schedule onto, which requires it to know the predicate and priority functions.
+-  increases the race condition between pod preemption and pending pod scheduling.
+
+Another option is for the scheduler to send the pending pod to a node without doing any preemption, and relying on the kubelet to do the preemption(s). Similar to the rescheduler option, this option requires replicating the preemption and scheduling logic. Kubelet already has the logic to evict pods when a node is under resource pressure, but this logic is much simpler than the whole scheduling logic that considers various scheduling parameters, such as affinity, anti-affinity, PodDisruptionBudget, etc. That is why we believe the scheduler is the right component to perform preemption.
+
+## Preemption order
+
+An alternative to preemption by priority and breaking ties with QoS which was proposed earlier, is to preempt by QoS first and break ties with priority. We believe this could cause confusion for users and might reduce cluster utilization. Imagine the following scenario:
+
+-  User runs a web server with a very high priority and is willing to give as much resources as possible to this web server. So, the users chooses a reasonable "requests" for resources and does not set any "limits" to let the web server use as much resources as it needs.
```

Comment:
This example is resource specific.  It's really only meaningful with memory.  Cpu in the scenario will still be limited.  Maybe clarify that it's memory only.